tachyon/zk/plonk/permutation/permutation_proving_key.h

```diff
@@ -0,0 +1,80 @@
+#ifndef TACHYON_ZK_PLONK_PERMUTATION_PERMUTATION_PROVING_KEY_H_
+#define TACHYON_ZK_PLONK_PERMUTATION_PERMUTATION_PROVING_KEY_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/buffer/copyable.h"
+#include "tachyon/math/polynomials/univariate/univariate_evaluations.h"
+#include "tachyon/math/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon {
+namespace zk {
+
+template <typename F, size_t MaxDegree>
+class PermutationProvingKey {
+ public:
+  using DensePoly = math::UnivariateDensePolynomial<F, MaxDegree>;
+  using Evals = math::UnivariateEvaluations<F, MaxDegree>;
+
+  PermutationProvingKey() = default;
+
+  explicit PermutationProvingKey(const std::vector<Evals>& permutations,
+                                 const std::vector<DensePoly>& polys)
+      : permutations_(permutations), polys_(polys) {}
+
+  explicit PermutationProvingKey(std::vector<Evals>&& permutations,
```

Comment:
```suggestion
  PermutationProvingKey() = default;
  explicit PermutationProvingKey(const std::vector<Evals>& permutations,
                                 const std::vector<DensePoly>& polys)
      : permutations_(permutations), polys_(polys) {}
  explicit PermutationProvingKey(std::vector<Evals>&& permutations,
```

Reply:
I was thinking is it good to say about it, and you did LoL